public/app/features/datasources/components/picker/DataSourcePicker.tsx

```diff
@@ -233,8 +226,13 @@ export function DataSourcePicker(props: DataSourcePickerProps) {
           onKeyDown={onKeyDownInput}
           value={filterTerm}
           onChange={(e) => {
-            openDropdown();
```

Comment:
No need to open the dropdown, it is already open with mouse click, thus search input is visible.